pkgs/by-name/ki/kimai/package.nix

```diff
@@ -0,0 +1,61 @@
+{pkgs ? import <nixpkgs> {
+  inherit system;
+}, system ? builtins.currentSystem,
+config,
+lib,
+consoleCmd ? "kimai-console",
+databaseUrl ? "",
+mailerFrom ? "",
+mailerUrl ? "",
+appSecret ? "",
+corsAllowOrigin ? "",
+...
+}:
+
+let
+  # userCfg = config.services.kimai;
+  phpPackage = import ./composer2nix/default.nix {
+    inherit pkgs system lib;
+  };
+in
+phpPackage.override {
+  postInstall = ''
+    # ensure the dirs exist, just in case,
+    # move any initialized data to "sample" to make space for symlinks.
+    # Finally, symlink to a directory outside of the nix store so that kimai
+    # may write to them without issue.
+    # These symlinks are created in the nix store but their targets need to be created later on,
+    # when the service starts.
+
+    mkdir -p ./var/cache
+    mkdir -p ./var/log
+    mkdir -p ./var/data
+    mv ./var/cache ./var/cache.sample
+    mv ./var/log ./var/log.sample
+    mv ./var/data ./var/data.sample
+    ln -s /var/lib/kimai/var/cache ./var/cache
+    ln -s /var/lib/kimai/var/log ./var/log
+    ln -s /var/lib/kimai/var/data ./var/data
+    ln -s /var/lib/kimai/var/plugins ./var/plugins
```

Comment:
yeah this isn't great, but given that PHP's runtime thingy [symfony](https://symfony.com/) insists on adding run-time data to the installation directory.... this was the only way around it.

Reply:
Works fine though! Look at `systemd.services.phpfpm-kimai.preStart` for the matching directory management.